common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -3770,6 +3774,27 @@ object CelebornConf extends Logging {
       .checkValues(Set(ShuffleMode.HASH.name, ShuffleMode.SORT.name))
       .createWithDefault(ShuffleMode.HASH.name)
 
+  val CLIENT_PUSH_DYNAMIC_WRITE_MODE_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.client.spark.push.dynamicWriteMode.enabled")
+      .categories("client")
+      .doc("Whether to dynamically switch push write mode based on conditions.If true, " +
+        s"shuffle mode will be is determined by partition count," +
+        s"not ${SPARK_SHUFFLE_WRITER_MODE.key}")
+      .version("0.4.0")
```

Comment:
let's move to 0.5.0, the release manager is preparing voting for 0.4.0 now

Reply:
Updated